impl/sql/mutationstorage/queue.go

```diff
@@ -162,24 +162,24 @@ func (m *Mutations) HighWatermark(ctx context.Context, directoryID string, logID
 		`SELECT COUNT(*), COALESCE(MAX(T1.Time), ?) FROM 
```

Comment:
How about:

- `SELECT COUNT(*), COALESCE(MAX(T1.Time)+1, ?)`
- `start-1` -> `start` in line 169
- `return count, high, nil` in line 173

This way it will return `high=low` if the batch is empty, whereas the current implementation will always increase `high` for no reason.